mirror_base.h

```diff
@@ -139,23 +139,8 @@ namespace mirror
 	{
 		static TypeDesc* Get()
 		{
-			static PointerTypeDesc s_pointerTypeDesc(TypeDescGetter<std::remove_pointer<T>::type>::Get()); return &s_pointerTypeDesc;
-		}
-	};
-
-	template <typename T>
-	struct TypeDescGetter<T, void, std::enable_if_t<std::is_enum<T>::value>>
-	{
-		static TypeDesc* Get()
-		{
-			switch (sizeof(T))
-			{
-			case 1: return TypeDescGetter<int8_t>::Get();
```

Comment:
Previously, here the `typeDescGetter<int8_t>` do not exists yet. By moving the declaration of `struct TypeDescGetter<T, void, std::enable_if_t<std::is_enum<T>::value>>` we are sure all TypeDescGetter<T> will exists (for in8_t, 16, 32 and 64)